src/main/java/org/zalando/baigan/service/ConfigurationParser.java

```diff
@@ -0,0 +1,79 @@
+package org.zalando.baigan.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.zalando.baigan.model.Condition;
+import org.zalando.baigan.model.Configuration;
+import org.zalando.baigan.proxy.BaiganConfigClasses;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
+public class ConfigurationParser {
+
+    private final Logger LOG = LoggerFactory
+            .getLogger(ConfigurationParser.class);
+    final ObjectMapper objectMapper;
+    final BaiganConfigClasses baiganConfigClasses;
+
+    // TODO define package structure so this is not public
```

Comment:
Can we go one step further and have the whole `BaiganConfigClasses` mechanism hidden from library users? As far as I can tell in `ConfigurationBeanDefinitionRegistrar` the library is providing and configuring the respective (singleton) bean for the application already. What if we, for example, let the `ConfigurationParser` obtain the bean from the application context for example?

Reply:
Now that we have the ConfigurationParser, this should indeed be quite conveniently possible. I'll have a look into this.

Reply:
Top!